spinta/manifests/xsd/helpers.py

```diff
@@ -0,0 +1,749 @@
+import os
+from copy import deepcopy
+
+from lxml.etree import _Element
+
+from spinta.components import Context
+from lxml import etree, objectify
+from urllib.request import urlopen
+from pprint import pprint
+
+from spinta.utils.naming import to_property_name, to_model_name, Deduplicator
+
+# mapping of XSD datatypes to DSA datatypes
+# XSD datatypes: https://www.w3.org/TR/xmlschema11-2/#built-in-datatypes
+# DSA datatypes: https://atviriduomenys.readthedocs.io/dsa/duomenu-tipai.html#duomenu-tipai
+# todo finish mapping and make sure all things are mapped correctly
+DATATYPES_MAPPING = {
+    "string": "string",
+    "boolean": "boolean",
+    "decimal": "number",
+    "float": "number",
+    "double": "number",
+
+    # Duration reikia mapinti su number arba integer, greičiausiai su integer ir XML duration
+    # reikšmė konvertuoti į integer reikšmę nurodant prepare funkciją, kuri konveruoti duration
+    # į integer, papildomai property.ref stulpelyje reikia nurodyti laiko vienetus:
+    # https://atviriduomenys.readthedocs.io/dsa/vienetai.html#laiko-vienetai
+    # todo add prepare functions
+    "duration": "",
+
+    "dateTime": "datetime",
+    "time": "time",
+    "date": "date",
+    "gYearMonth": "date;ref:M",
+    "gYear": "date;ref:Y",
+    "gMonthDay": "string",
+    "gDay": "string",
+    "gMonth": "string",
+    "hexBinary": "string",
+    "base64Binary": "string",
+    "anyURI": "uri",
+    "QName": "string",
+    "NOTATION": "string",
+    "normalizedString": "string",
+    "token": "string",
+    "language": "string",
+    "NMTOKEN": "string",
+    "NMTOKENS": "string",
+    "Name": "string",
+    "NCName": "string",
+    "ID": "string",
+    "IDREF": "string",
+    "IDREFS": "string",
+    "ENTITY": "string",
+    "ENTITIES": "string",
+    "integer": "integer",
+    "nonPositiveInteger": "integer",
+    "negativeInteger": "integer",
+    "long": "integer",
+    "int": "integer",
+    "short": "integer",
+    "byte": "integer",
+    "nonNegativeInteger": "integer",
+    "unsignedLong": "integer",
+    "unsignedInt": "integer",
+    "unsignedShort": "integer",
+    "unsignedByte": "integer",
+    "positiveInteger": "integer",
+    "yearMonthDuration": "integer",
+    "dayTimeDuration": "integer",
+    "dateTimeStamp": "datetime",
+    "": "string",
+
+}
+
+"""
+format of custom types:
+{
+    "type_name": {
+        "base": "type_base"
+    }
+}
+"""
+
+
+class Deduplicator2:
```

Comment:
This should go to `spinta.utils.naming` and I think, we already have identical class named [spinta.utils.naming.Deduplicator](https://github.com/atviriduomenys/spinta/blob/6930b1af53fcf2b8e044b8dd797df5218c328c19/spinta/utils/naming.py#L53).

Reply:
this class isn't identical. Class Deduplicator in `spinta.utils.naming` has a class variable `_names` and `Deduplicator2` has this variable as an instance variable. This is because we might want to have different deduplicators with their own `_name` sets when there is a recursion, so that it would deduplicate only names of those properties, which are in the same model.

Reply:
`Deduplicator._names` is not a class attribute, it is an instance attribute.

Because at class level it only has type annotation, but value is not set, value is set at instance level in `__init__`. That makes these two classes identical, only `Deduplicator2` does not have type annotations, which is a bad thing, all variables should have type annotations, because some time in the future, there are plans to run static code checker mypy to improve code quality.

Reply:
pataisyta :)